dbms/tests/queries/0_stateless/00753_alter_destination_for_storage_buffer.sql

```diff
@@ -0,0 +1,32 @@
+DROP TABLE IF EXISTS test.dst;
+DROP TABLE IF EXISTS test.buffer;
+SET send_logs_level = 'error';
+
+CREATE TABLE test.dst (x UInt64, y UInt64) ENGINE = MergeTree ORDER BY tuple();
+CREATE TABLE test.buffer (x UInt64, y UInt64) ENGINE = Buffer(test, dst, 1, 99999, 99999, 1, 1, 99999, 99999);
+
+INSERT INTO test.buffer VALUES (1, 100);
+INSERT INTO test.buffer VALUES (2, 200);
+INSERT INTO test.buffer VALUES (3, 300);
+SELECT 'init';
+SELECT * FROM test.dst ORDER BY x;
+SELECT '-';
+SELECT * FROM test.buffer ORDER BY x;
+
+ALTER TABLE test.dst DROP COLUMN x, MODIFY COLUMN y String, ADD COLUMN z String DEFAULT 'DEFZ';
+
+INSERT INTO test.buffer VALUES (4, 400);
```

Comment:
There was an exception at this line:

```
<Error> executeQuery: Code: 16, e.displayText() = DB::Exception: There is no column with name x. There are columns: y, z: (when looking at destination table test.dst), e.what() = DB::Exception (from 127.0.0.1:40774) (in query: INSERT INTO test.buffer VALUES), Stack trace:

/home/user/work/ClickHouse/build/dbms/programs/clickhouse-server(DB::Exception::Exception(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, int)+0x22) [0x2a67762]
/home/user/work/ClickHouse/build/dbms/programs/clickhouse-server(DB::ITableDeclaration::check(DB::Block const&, bool) const+0x3bc) [0x58918fc]
/home/user/work/ClickHouse/build/dbms/programs/clickhouse-server(DB::BufferBlockOutputStream::write(DB::Block const&)+0x12f) [0x5d00a9f
```

With my changes this line is successfully executed and causes inserting new row

y | z
--|--
400 | DEFZ

to the table `test.dst`.